main.js

```diff
@@ -286,7 +286,7 @@ app.post('/getIngredientForCustomRecipe', function (req, res, next) {
 
```

Comment:
The `getIngredientForCustomRecipe` function does three different things, with three nested database queries.
1. Find the name of an ingredient using its ID
1. Find the ingredient's ethical problem (if any)
1. Find the ingredient's alternatives (if any)

It's become long and hard to understand. Here is what I suggest: 
1. The first database call can be removed. The client should already know the name of an ingredient.
1. Break the second task into its own separate end point. For example, `/getEthicalProblemForIngredientId`.
1. Remove the code for task 3 altogether and use the `/getAlternativesForIngredientId` instead.
1. Change the client so that it makes two separate requests to `/getEthicalProblemForIngredientId` and `/getAlternativesForIngredientId` respectively.

Reply:
I'll work on these changes in my code smells branch.